corehq/apps/reports/filters/api.py

```diff
@@ -197,3 +223,194 @@ def get(self, request, domain):
 
 class DeviceLogIds(DeviceLogFilter):
     field = 'device_id'
+
+
+@require_POST
+@toggles.COPY_CASES.required_decorator()
+@require_permission(HqPermissions.edit_data)
+@requires_privilege(privileges.CASE_COPY)
+@location_safe
+def copy_cases(request, domain, *args, **kwargs):
+    body = json.loads(request.body)
+
+    case_ids = body.get('case_ids')
+    if not case_ids:
+        return JsonResponse({'error': _("Missing case ids")}, status=400)
+
+    new_owner = body.get('owner_id')
+    if not new_owner:
+        return JsonResponse({'error': _("Missing new owner id")}, status=400)
+
+    censor_data = {
+        prop['name']: prop['label']
+        for prop in body.get('sensitive_properties', [])
+    }
+
+    case_copier = CaseCopier(
+        domain,
+        to_owner=new_owner,
+        censor_data=censor_data,
+    )
+    case_id_pairs, errors = case_copier.copy_cases(case_ids)
+    count = len(case_id_pairs)
+    return JsonResponse(
+        {'copied_cases': count, 'error': errors},
+        status=400 if count == 0 else 200,
+    )
+
+
+@dataclass
+class UserDuck:
```

Comment:
This class is also defined [here](https://github.com/dimagi/commcare-hq/blob/96044ac2d2d2beddf67a097c7a0cbc63996d7aaa/corehq/apps/hqcase/case_helper.py#L192) in `case_helper`, but is currently inside a private function. You could move the dataclass out of the private method and then use it here.

Reply:
Ah, thanks for pointing that out!